src/Compilers/Core/Portable/InternalUtilities/ReflectionUtilities.cs

```diff
@@ -73,5 +75,34 @@ public static T CreateDelegate<T>(this MethodInfo methodInfo)
 
             return (T)(object)methodInfo.CreateDelegate(typeof(T));
         }
+
+        public static T InvokeConstructor<T>(this ConstructorInfo constructorInfo, params object[] args)
+        {
+            if (constructorInfo == null)
+            {
+                return default(T);
+            }
+
+            try
+            {
+                return (T)constructorInfo.Invoke(args);
+            }
+            catch (TargetInvocationException e)
+            {
+                ExceptionDispatchInfo.Capture(e.InnerException).Throw();
+                Debug.Assert(false, "Unreachable");
```

Comment:
You're totally right. @jaredpar should have done that before I copied and pasted his code. :smiling_imp:

Easy enough change to make.